### PR TITLE
Fix regressions in user management

### LINKED ISSUE #1993: UI - No Checkbox to Grant `USER_PERMISSION_UPDATE`
### Bug Description

In the user creation page, list the scopes that could be applied to a user in the form of Checkboxes. In that list of scopes, we allow users to be granted "create role", which maps to `USER_PERMISSION_CREATE` and "view role", which maps to `USER_PERMISSION_READ`; however, there is not checkbox that maps to `USER_PERMISSION_UPDATE`.

This means that any user created through the UI is unable to be granted the ability to modify other users' permissions. The workaround for this is to use the API to modify permissions.

#### Steps to Reproduce

1. Go to the create user page
2. Create user, give scopes for create and view roles and users
3. Log in as that new user, create another user and give them permissions to anything
4. Save that user
5. Then try to edit the permissions of that user that was just created.

### Expected behavior

We should have a checkbox that allows users to edit the permissions applied to a user.

### Screenshots

image

### Environment

* Version: 2.1.0
* OS:
* Python Version:
* Docker Version:

### Additional context

_Add any other context about the problem here._

### PR DESCRIPTION
**Pull request by seanpreston**
Closes #1993 

### Code Changes

* [x] Adds logging output to `verify_oauth_token` logic used to verify scopes in the API
* [x] Adds the option to assign users the `user:update` scope
* [x] Adds the option to assign users the `user:reset-password` scope
* [x] Adds the option to assign users the `user_permission:update` scope
* [x] Updates the copy of the existing scopes such that `role` -> `user permission`

Screenshot 2022-12-06 at 17 46 22

### Steps to Confirm

* [ ] Run `nox -s dev`
* [ ] Navigate to the user management page
* [ ] Create a new user
* [ ] Confirm that `Update user permissions` shows as an option to assign to users
* [ ] Assign that permission
* [ ] Login as the new user
* [ ] Go to the user management page
* [ ] Confirm that you can update permissions for a user that is not the root user

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  * [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  * [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Review
- [ ] @conceptualshark: can you please confirm this new copy works for you?
- [ ] @RobertKeyser / @nicolas-ethyca: keeping you cc'd

### CODE REVIEW

File: clients/admin-ui/src/constants.ts [hunk 1]
```diff
@@ -77,11 +77,19 @@ export const USER_PRIVILEGES: UserPrivileges[] = [
     scope: "user:create",
   },
   {
-    privilege: "Create roles",
+    privilege: "Reset their own password",
+    scope: "user:reset-password",
```
Comment (seanpreston) [inline clients/admin-ui/src/constants.ts:81]: This will allow users to have the options to reset their own passwords.

Comment (RobertKeyser) [reply clients/admin-ui/src/constants.ts:85]: Did they not have this before? Is there any way to have this default to `true`?

**Comment by TheAndrewJackson** [comment]
@seanpreston In my testing I found that a user is able to edit another users permissions but not their own. Is this the expected behavior? Here is a video showing what I found:


https://user-images.githubusercontent.com/17103888/206247158-e3fcc71a-16cc-46d0-b78f-bfb797e218e8.mov

I accidentally unchecked "Read User Permissions". The same thing happens if it's checked.

**Comment by seanpreston** [comment]
> @seanpreston In my testing I found that a user is able to edit another users permissions but not their own. Is this the expected behavior?

This issue is known logged [here](https://github.com/orgs/ethyca/projects/18/views/2?filterQuery=is%3Aissue+is%3Aopen+epic%3A%22User+Management%22). The error there is that the no HTTP request is being sent when users try to edit their own attributes.

**Review by TheAndrewJackson — APPROVED** [review]
Looks good! Everything worked for me other than the known issue

File: clients/admin-ui/src/constants.ts (hunk 1 above)
Comment (seanpreston) [reply clients/admin-ui/src/constants.ts:85]: We can do that too in a follow-up 👍

**Pull request merged**